dso-l2/src/main/java/com/tc/objectserver/impl/ServerPersistenceVersionChecker.java

```diff
@@ -19,52 +19,30 @@
 
 package com.tc.objectserver.impl;
 
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
-
 import com.tc.objectserver.persistence.ClusterStatePersistor;
 import com.tc.util.ProductInfo;
 import com.tc.util.version.Version;
-import com.tc.util.version.VersionCompatibility;
 
 /**
  * @author tim
  */
 public class ServerPersistenceVersionChecker {
-  private static final Logger LOGGER = LoggerFactory.getLogger(ServerPersistenceVersionChecker.class);
-  private final ClusterStatePersistor clusterStatePersistor;
   private final ProductInfo productInfo;
-  private final VersionCompatibility versionCompatibility;
 
-  public ServerPersistenceVersionChecker(ClusterStatePersistor clusterStatePersistor) {
-    this(clusterStatePersistor, ProductInfo.getInstance(), new VersionCompatibility());
+  ServerPersistenceVersionChecker() {
+    this(ProductInfo.getInstance());
   }
 
-  ServerPersistenceVersionChecker(ClusterStatePersistor clusterStatePersistor, ProductInfo productInfo,
-                                  VersionCompatibility versionCompatibility) {
-    this.clusterStatePersistor = clusterStatePersistor;
+  ServerPersistenceVersionChecker(ProductInfo productInfo) {
     this.productInfo = productInfo;
-    this.versionCompatibility = versionCompatibility;
-  }
-
-  private boolean checkVersion(Version persisted, Version current) {
-    if (persisted != null) { return versionCompatibility.isCompatibleServerPersistence(persisted, current); }
-    return true;
   }
 
-  public void checkAndSetVersion() {
+  void checkAndBumpPersistedVersion(ClusterStatePersistor clusterStatePersistor) {
```

Comment:
Changed method name as suggested by Myron. Also refactored to make the side effect on clusterStatePersistor more explicit by passing it in as a method argument rather than a field.